src/main/kotlin/device/application/CreateDevice.kt

```diff
@@ -0,0 +1,32 @@
+package device.application
+
+import device.domain.Device
+import device.domain.DeviceIdentifier
+import device.domain.DeviceName
+import device.domain.DeviceRepository
+import device.domain.exceptions.AlreadyExistingDevice
+import device.domain.exceptions.InvalidUUIDException
+
+class CreateDevice(private val repository: DeviceRepository) {
+
+    /***
+     * Creates a new device from primitives and saves it to the DeviceRepository
+     * @throws InvalidUUIDException if the provided id is not a valid UUIDv4 identifier
+     * @throws AlreadyExistingDevice if a new device already exists
+     * @return The device instance that was saved inside the repository
+     */
+
+    fun create(id: String, name: String) = create(DeviceIdentifier(id), DeviceName(name))
+
+    /***
+     * Creates a new device from domain objects and saves it to the DeviceRepository
+     * @throws AlreadyExistingDevice if a new device already exists
+     * @return The device instance that was saved inside the repository
+     */
+
+    fun create(id: DeviceIdentifier, name: DeviceName): Device {
```

Comment:
I was thinking that the best visibility for this type of functions should be "internal", since it is great to have this functions that work with domain objects from inside the internals of the domain logic, but we don't want the end user to have this functions "public" to the end user.

If the function is internal we don't need the documentation for it.

I will add an issue for adding the use cases with primitives (like you did) and making the domain specific functions internal.